src/polymer/polymer-element.ts

```diff
@@ -262,9 +262,9 @@ function resolveElement(
     scannedElement: ScannedPolymerElement, document: Document): PolymerElement {
   const element = new PolymerElement();
   element.privacy = scannedElement.privacy;
-  applySuperClass(element, scannedElement, document);
-  applyMixins(element, scannedElement, document);
   applySelf(element, scannedElement, document);
+  applyMixins(element, scannedElement, document);
+  applySuperClass(element, scannedElement, document);
```

Comment:
I think there are three arguments in favor of this approach, in descending order of desirability:

With the current implementation of properties and friends as simple arrays, the implementation of overriding is tricky to do in a way that isn't O(N^2). Doing it this way is simple to read, simple to write, and IMO simple to understand once you accept that we go in leaf-down order rather than root-up. I think we're still in a good position to take advantage of caching of intermediate values, as we're doing here by mixing mixins into mixins.

Another issue is that I'd like to move as much of this code as possible up to Element and even to Class (once that lands), but if we go from top of the prototype chain down then that leaves us in a somewhat awkward position with behaviors. Maybe with the unification of behaviors and mixins in #630 we could do something there, but does seem a bit tricky.

And a yet smaller issue, the ordering of properties in this implementation is more legible, as local properties come first, in declaration order, followed by inherited properties. This is minor, and we could modify the code in generateAnalysis to do this sorting, the editor service does it.